src/sql/util.py

```diff
@@ -157,6 +157,84 @@ def show_deprecation_warning():
     )
 
 
+def check_duplicate_arguments(
+    magic_execute, cmd_from, args, allowed_duplicates
+) -> bool:
+    """
+    Raises UsageError when duplicate arguments are passed to magics.
+    Returns true if no duplicates in arguments or aliases.
+    """
+
+    aliased_arguments = {}
+    unaliased_arguments = []
+
+    for decorator in magic_execute.decorators:
+        assert isinstance(decorator.args, tuple)
+        dec_args = decorator.args
+        if len(dec_args) > 1:
+            aliased_arguments[dec_args[0]] = dec_args[1]
+        else:
+            if dec_args[0].startswith("--") or dec_args[0].startswith("-"):
+                unaliased_arguments.append(dec_args[0])
+
+    if len(args) <= 1:
+        return True
+
+    args = [arg for arg in args if arg.startswith("--") or arg.startswith("-")]
+
+    single_hyphen_opts = set()
+    double_hyphen_opts = set()
+
+    for arg in args:
+        if arg.startswith("--"):
+            double_hyphen_opts.add(arg)
+        elif arg.startswith("-"):
+            single_hyphen_opts.add(arg)
+
+    duplicate_args = []
+    visited_args = set()
+    for arg in args:
+        if arg not in allowed_duplicates:
+            if arg not in visited_args:
+                visited_args.add(arg)
+            else:
+                duplicate_args.append(arg)
+
+    alias_pairs_present = [
+        (opt, aliased_arguments[opt])
+        for opt in single_hyphen_opts
+        if opt in aliased_arguments
+        if aliased_arguments[opt] in double_hyphen_opts
+    ]
+
+    error_message = ""
+    if duplicate_args:
+        duplicates_error = (
+            f"Duplicate arguments in %{cmd_from}. "
+            "Please use only one of each of the following: "
+            f"{', '.join(sorted(duplicate_args))}. "
+        )
+    else:
+        duplicates_error = ""
+
+    if alias_pairs_present:
```

Comment:
This function is a bit hard to understand in general. The following improvements can be made:
1. Add more relevant comments
2. Break down into smaller internal functions
3. Better variable naming

Reply:
Improved readability in [this](https://github.com/ploomber/jupysql/pull/908/commits/b5db9f347004e94f68d64f3e76a1244fd895c0a8) commit